src/kafka-extensions/registry.ts

```diff
@@ -0,0 +1,181 @@
+import * as vscode from "vscode";
+import { KafkaConfig } from "kafkajs";
+import { Cluster, ConnectionOptions, createDefaultKafkaConfig as createDefaultKafkaConfig } from "../client/client";
+import { ClusterSettings } from "../settings/clusters";
+import { collectDefaultClusters } from "../wizards/clusters";
+import { ClusterProviderParticipant, KafkaExtensionParticipant } from "./api";
+
+/**
+ * Cluster provider is used to:
+ *
+ *  - collect clusters (eg: create a cluster from a wizard, import clusters from a repository, ...)
+ * and add thento the Kafka Explorer.
+ *  - create a Kafka client from a complex process (eg : use SSO to connect to the cluster)
+ *
+ * Implementing a cluster provider in custom vscode extension is done in 2 steps:

```

Comment:
Implementing a cluster provider in a 3rd party extension is done in 2 steps:

Reply:
fixed